### ISSUE TITLE
`Invoke-OAIChatCompletion` has a `-Tools` param - should work the same as `New-Agent`

### ISSUE THREAD

**Issue by dfinke** [dfinke/PSAI#81]
For `Invoke-OAIChatCompletion -Tools` you can pass in functions htat have doen `Register-Tool`. 
The param should have the same behavior `New-Agent -Tools` has.

Also an opportunity to cosider centralizing that functionality.

**Comment by Agazoth** [comment]
Tool calls are currently handled in the PrintResponse method of New-Agent. If Invoke-OAIChatCompletion should have the same functionality as New-Agent, the method needs to be implemented there too.

I think we need to re-think where to insert the tool running functionality, so it works with all kinds of calls across providers.

**Comment by dfinke** [comment]
Yes. Looking at the OpenAI examples, I want to be able to as easy as possible do a one to one port to PowerShell and get the example working.

Invoke-OAIChatCompletion needs work - don't know if a new function may be needed. Depending on that, "normalizing" tools across providers is needed. I wonder how Groq does it?
